sql/core/src/main/scala/org/apache/spark/sql/execution/dynamicpruning/PartitionPruning.scala

```diff
@@ -196,6 +198,25 @@ object PartitionPruning extends Rule[LogicalPlan] with PredicateHelper {
     case _ => false
   }
 
+  /**
+   * For some filtering side can not broadcast by join type but can broadcast by size,
+   * then we should not consider reuse broadcast only, for example:
+   * Left outer join and left side very small.
+   */
+  private def isReuseBroadcastOnly(
+      canBuildBroadcast: Boolean,
+      hasBenefit: Boolean,
+      hintToBroadcastPruningSide: Boolean,
+      pruningPlan: LogicalPlan,
+      filteringPlan: LogicalPlan): Boolean = {
+    if (canBuildBroadcast) {
+      !hasBenefit || SQLConf.get.dynamicPartitionPruningReuseBroadcastOnly
+    } else {
+      !(hasBenefit && canBroadcastBySize(filteringPlan, SQLConf.get) &&
+        !hintToBroadcastPruningSide && !canBroadcastBySize(pruningPlan, SQLConf.get))
```

Comment:
`!canBroadcastBySize(pruningPlan, SQLConf.get)` do we need this condition?

Reply:
Yes. To avoid driver OOM.